File: php/jsonproto/app/Benchmark/Dtos/TransportDto.php
<?php

namespace App\Benchmark\Dtos;

use Carbon\CarbonImmutable;
use DateTimeImmutable;

class TransportDto
{
    public function __construct(
        public string $id,
        public string $code,
        public string $externalId,
        public DateTimeImmutable|CarbonImmutable $plannedStartAt,
        public DateTimeImmutable|CarbonImmutable $plannedEndAt,
        public DateTimeImmutable|CarbonImmutable $actualStartedAt,
        public DateTimeImmutable|CarbonImmutable $actualEndedAt,
        public float $cost,
        public float $travelTime,
        /** @var PackageDto[] $packages */
        public array $packages,
    )
    {}
}

File: java/src/main/java/io/belac/blog/dtos/TransportDto.java
package io.belac.blog.dtos;

import java.time.OffsetDateTime;
import java.util.List;

public record TransportDto(
        String id,
        String code,
        String externalId,
        OffsetDateTime plannedStartAt,
        OffsetDateTime plannedEndAt,
        OffsetDateTime actualStartedAt,
        OffsetDateTime actualEndedAt,
        Double cost,
        Double travelTime,
        List<PackageDto> packages
) {
}
